pkg/store/tsdb/influx/datastore.go

```diff
@@ -69,10 +68,62 @@ func (ds DataStore) WriteHeartBeats(username string, heartBeats []models.HeartBe
 		}
 		bp.AddPoint(pt)
 	}
-
 	return ds.Write(bp)
 }
-
 func toTime(t float64) time.Time {
 	return time.Unix(int64(t), 0)
 }
+
+// QueryHeartBeats gets all heartbeats in a given time range.
+func (ds DataStore) QueryHeartBeats(begin time.Time, end time.Time) ([]models.HeartBeatFrontModel, error) {
+	queryString := fmt.Sprintf("SELECT * FROM %s WHERE time < %s AND time > %s", constants.MeasurementName, begin.Format(time.RFC3339), end.Format(time.RFC3339))
+	responses, err := ds.Query(client.NewQuery(queryString, ds.DBName, "s"))
+	if err != nil {
+		return nil, err
+	}
+	heartbeats := []models.HeartBeatFrontModel{}
+	for _, result := range responses.Results {
+		for _, row := range result.Series {
+			for _, onepoint := range row.Values {
+				hb := models.HeartBeatFrontModel{}
+				//TODO: currently some fields of HB is not stored into influxDB
+				//so when we retrieve it back we could only get empty placeholders.
+				//someday we may be able to store them.
+				hb.Dependencies = []string{}
+
+				for i := 0; i < len(onepoint); i++ {
+					switch row.Columns[i] {
```

Comment:
Actually things like this can be generated, it's just influxdb does not have a generator/struct binding like most SQL databases ... the generated code/binding runtime is just like this hand written one.

Reply:
How to generate this one? Is there any tutorials about this?

Reply:
the easiest way is to use reflect, you can see database orm library like gorm https://github.com/jinzhu/gorm https://github.com/jmoiron/sqlx etc. though if it is the only place then we can just keep it as it is

Reply:
We are using gorm, but I guess it is not for TSDBs like influxDB...